server/routes.ts

```diff
@@ -113,6 +113,12 @@ export default function routes(app: Express) {
   router.route('/workspaces/:id').delete(isAuth('admin'), workspace.delete);
   router.route('/workspaces/duplicate').post(isAuth(), user.duplicateWS);
 
+  // workspace share apis
+  router.route('/workspaces/share/:id').get(isAuth(), workspace.getShare);
```

Comment:
look like every one can get the share ? not ony the creator and dest ?

Reply:
All the share APIs query to database with `creatorId` identified from token and hence will only work if the `creatorId` matches with the workspaceId